lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart

```diff
@@ -225,19 +198,9 @@ class MailboxDashBoardController extends ReloadableController {
         });
   }
 
-  void _registerSearchFocusListener() {
-    searchFocus.addListener(() {
-      final hasFocus = searchFocus.hasFocus;
-      final query = searchQuery?.value;
-      log('MailboxDashBoardController::_registerSearchFocusListener(): hasFocus: $hasFocus | query: $query');
-      if (!hasFocus && (query == null || query.isEmpty)) {
-        disableSearch();
-      }
-    });
-  }
-
   void _getUserProfile() async {
     userProfile.value = sessionCurrent != null ? UserProfile(sessionCurrent!.username.value) : null;
+    searchController.userProfile.value = userProfile.value;
```

Comment:
I think we accepted: we don't follow this way

Reply:
It seems like search_controller will binding before mailbox_dashboard_controller.dart (you can check it in [here](lib/features/mailbox_dashboard/presentation/mailbox_dashboard_bindings.dart) ). So we can't use this way talk before. Now I added `userProfile` and `accountId` as parameter of functions. when we use functions just add it as parameter.

Reply:
Why we need to binding `SearchController` before `mailbox_dashboard_controller`? What is your strategy when separate the `controllers`?

Reply:
I think you had an answer ( like below https://github.com/linagora/tmail-flutter/pull/683#discussion_r906734865). In the future, I want to move feature search to 1 module ( easy to re-use and maintain ). So  I think we should let other controllers depend on the searchcontroller, not the other way around.